### ISSUE TITLE
angular 12 error 'mgl-map' is not a known element using VSCode, angular language service

### ISSUE THREAD

**Issue by jpike88** [Wykks/ngx-mapbox-gl#330]
We import a range of libraries in our app, and this is the only one showing this problem.

We have strict template checked enabled, and are using Ivy.

<img width="938" alt="Screenshot 2021-09-10 at 3 05 14 PM" src="https://user-images.githubusercontent.com/9585787/132821359-500634b0-a868-4a34-9d0a-7a99e76e0361.png">

See above image.

We have tried the default way of including it in our app module, and the forRoot method described at the bottom of the readme.

When we build the app, it seems some sort of functionality is kicking in, but no map is showing.

<img width="1023" alt="Screenshot 2021-09-10 at 3 06 54 PM" src="https://user-images.githubusercontent.com/9585787/132821555-9c546d2b-402c-4f61-816e-7e4618d6783e.png">

This is not something that I should be putting on stack overflow, I'm an angular veteran and know that this library is doing something weird that is throwing off the language service. The issue should be resolved here.

**Comment by dmytro-gokun** [comment]
@jpike88 I totally agree, the issue (if it really exists) should be resolved here. We need two things for that:
1) reproduction code;
2) a PR to fix the issue.

You are welcome to help with both.

**Comment by jpike88** [comment]
update: I made a simple wrapper in-house, the map works fine, no type or build errors. so the issue is definitely to do with how the module interacts with angular's language service.

**Comment by dmytro-gokun** [comment]
@jpike88 Great, now you just need to provide a reproduction and a PR. Thanks in advance!

**Comment by mendhak** [comment]
@jpike88  could you share your wrapper or any other details?  We're running into this issue as well (Angular 14) and I can't figure out what's missing.  The basic example given in the README doesn't compile.
